cli/f3.go

```diff
@@ -70,6 +75,243 @@ var (
+								if err != nil {
+									return fmt.Errorf("parsing instance: %w", err)
+								}
+								if instance > progress.ID {
+									// TODO: Technically we can return power table for instances ahead as long as
+									//       instance is within lookback. Implement it.
+									return fmt.Errorf("instance is ahead the current instance in progress: %d > %d", instance, progress.ID)
+								}
+							} else {
+								instance = progress.ID
+							}
+
+							ltsk, expectedPowerTableCID, err := f3GetPowerTableTSKByInstance(cctx.Context, api, instance)
+							if err != nil {
+								return fmt.Errorf("getting power table tsk for instance %d: %w", instance, err)
+							}
+
+							var result = struct {
+								Instance   uint64
+								FromEC     bool
+								PowerTable struct {
+									CID         string
+									Entries     gpbft.PowerEntries
+									Total       gpbft.StoragePower
+									ScaledTotal int64
+								}
+							}{
+								Instance: instance,
+								FromEC:   cctx.Bool(f3FlagPowerTableFromEC.Name),
+							}
+							if result.FromEC {
+								result.PowerTable.Entries, err = api.F3GetECPowerTable(cctx.Context, ltsk)
+							} else {
+								result.PowerTable.Entries, err = api.F3GetF3PowerTable(cctx.Context, ltsk)
+							}
+							if err != nil {
+								return fmt.Errorf("getting f3 power table at instance %d: %w", instance, err)
+							}
+
+							pt := gpbft.NewPowerTable()
+							if err := pt.Add(result.PowerTable.Entries...); err != nil {
+								// Sanity check the entries returned by the API.
+								return fmt.Errorf("retrieved power table is not valid for instance %d: %w", instance, err)
+							}
+							result.PowerTable.Total = pt.Total
+							result.PowerTable.ScaledTotal = pt.ScaledTotal
+
+							actualPowerTableCID, err := certs.MakePowerTableCID(result.PowerTable.Entries)
+							if err != nil {
+								return fmt.Errorf("gettingh power table CID at instance %d: %w", instance, err)
+							}
+							if !expectedPowerTableCID.Equals(actualPowerTableCID) {
+								return fmt.Errorf("expected power table CID %s at instance %d, got: %s", expectedPowerTableCID, instance, actualPowerTableCID)
+							}
+							result.PowerTable.CID = actualPowerTableCID.String()
+
+							output, err := json.MarshalIndent(result, "", "  ")
+							if err != nil {
+								return fmt.Errorf("marshalling f3 power table at instance %d: %w", instance, err)
+							}
+							_, _ = fmt.Fprint(cctx.App.Writer, string(output))
+							return nil
+						},
+					},
+					{
+						Name:    "get-proportion",
+						Aliases: []string{"gp"},
+						Usage: `Gets the total proportion of power for a list of actors at a given instance.
+
+The instance may be specified as the first argument. If unspecified,
+the latest instance is used.
+
+The list of actors may be specified as Actor ID or miner address, space
+separated, pied to STDIN. Example:
+  $ echo "1413 t01234 f12345" | lotus f3 powertable get-proportion 42`,
```

Comment:
wouldn't it be easier to just accept these as trailing args? then make the `[instance]` an `[--instance id]`?

Reply:
All sub commands here take instace as the first arg. I wanted to keep to that. Happy to change it.